src/Polytopes/Cone.jl

```diff
@@ -0,0 +1,112 @@
+@doc Markdown.doc"""
+    Cone(Rays)
+
+     A polyhedral cone, not necessarily pointed, defined by the positive hull
+     of the rays, Rays.
+"""
+struct Cone #a real polymake polyhedron
+    pm_cone::Polymake.BigObjectAllocated
+end
+function Cone(Rays::Union{Oscar.MatElem,AbstractMatrix})
+    Cone(Polymake.polytope.Cone{Polymake.Rational}(
+        RAYS = matrix_for_polymake(Rays),
+    ))
+end
+
+"""
+    pm_polytope(C::Cone)
+
+Get the underlying polymake `Cone`.
+"""
+pm_cone(C::Cone) = C.pm_cone
+
+struct ConeRayIterator
+    cone::Cone
+end
+
+function Base.iterate(iter::ConeRayIterator, index = 1)
+    rays = pm_cone(iter.cone).RAYS
+    if size(rays, 1) < index
+        return nothing
+    end
+
+    return (rays[index, :], index + 1)
+end
+Base.eltype(::Type{ConeRayIterator}) = Polymake.VectorAllocated{Polymake.Rational}
+Base.length(iter::ConeRayIterator) = n_rays(iter.cone)
+
+"""
+   n_rays(C::Cone)
+
+Returns the number of rays of the cone `C`.
+"""
+n_rays(C::Cone) = size(pm_cone(C).RAYS, 1)
+
+"""
+   rays(C::Cone)
+
+Returns the rays of a cone.
+"""
+rays(C::Cone) = ConeRayIterator(C)
+
+
+"""
+    positive_hull(generators::Union{Oscar.MatElem,AbstractMatrix})
+
+A polyhedral cone, not necessarily pointed, defined by the positive hull
+of the `generators`. Redundant rays are allowed in the generators.
+"""
+function positive_hull(generators::Union{Oscar.MatElem,AbstractMatrix})
+    # TODO: Filter out zero rows
+    C=Polymake.polytope.Cone{Polymake.Rational}(INPUT_RAYS =
+      matrix_for_polymake(remove_zero_rows(generators)))
+    Cone(C)
+end
+
+#Is there a Polymake.jl function for this?
+function pm_set_to_julia(S::Polymake.SetAllocated{Int64})
+    return(Array{Int64,1}(S))
```

Comment:
`return` is not a function, so you can (and for "idiomatic" Julia, should) remove the parenthesis here and in similar instances.
```suggestion
    return Array{Int64,1}(S)
```

Reply:
Also, this is yet another function which apparently is not exercised by the existing test suite.